core/object/object.cpp

```diff
@@ -679,6 +679,58 @@ bool Object::has_method(const StringName &p_method) const {
 	return false;
 }
 
+int Object::_get_method_argument_count_bind(const StringName &p_method) const {
+	return get_method_argument_count(p_method);
+}
+
+int Object::get_method_argument_count(const StringName &p_method, bool *r_is_valid) const {
+	if (p_method == CoreStringNames::get_singleton()->_free) {
+		if (r_is_valid) {
+			*r_is_valid = true;
+		}
+		return 0;
+	}
+
+	if (script_instance) {
+		bool valid = false;
+		int ret = script_instance->get_method_argument_count(p_method, &valid);
+		if (valid) {
+			if (r_is_valid) {
+				*r_is_valid = true;
+			}
+			return ret;
+		}
+	}
+
+	{
+		bool valid = false;
+		int ret = ClassDB::get_method_argument_count(get_class_name(), p_method, &valid);
+		if (valid) {
+			if (r_is_valid) {
+				*r_is_valid = true;
+			}
+			return ret;
+		}
+	}
+
+	const Script *scr = Object::cast_to<Script>(this);
+	if (scr != nullptr) {
+		bool valid = false;
+		int ret = scr->get_script_method_argument_count(p_method, &valid);
+		if (valid) {
+			if (r_is_valid) {
+				*r_is_valid = true;
+			}
+			return ret;
+		}
+	}
```

Comment:
Shouldn't inheritance be checked here since `Script::get_script_method_argument_count()` doesn't take into account base classes, unlike `ScriptInstance::get_method_argument_count()`? (This is quite confusing. See also #80418.)

Reply:
Good point, I based this on how methods like `get_method_info` and `has_method` work in GDScript and Mono for the Script and ScriptInstance cases respectively, but open to other suggestions, can make this one iterate without trouble 🙂

Reply:
Also realized this is called on the class itself, got it confused with the static part in the case above, so should probably use `script` instead of casting itself, unsure though

Will add both for comparison